### DISCUSSION TITLE
I cannot start the server

### DISCUSSION THREAD

**Discussion by joehil** [e154/smart-home#261] [Q&A]
Hello Alexey,

I really like the idea of your smart-home application. But unfortunatelly I cannot get it to work. When I start the server I get the following messages:

database connected to dbname=smart_home user=smart_home password=***** host=127.0.0.1 port=5432 sslmode=disable
database version 15.5 (Debian 15.5-0+deb12u1)
database timezone: Europe/Berlin
INFO    scripts                 scripts/scripts.go:107 >        register structure: 'Log'
INFO    scripts                 scripts/scripts.go:119 >        register function: 'ExecuteSync'
INFO    scripts                 scripts/scripts.go:119 >        register function: 'ExecuteAsync'
INFO    scripts                 scripts/scripts.go:119 >        register function: 'Encrypt'
INFO    scripts                 scripts/scripts.go:119 >        register function: 'Decrypt'
INFO    scripts                 scripts/scripts.go:107 >        register structure: 'Storage'
INFO    scripts                 scripts/scripts.go:107 >        register structure: 'http'
INFO    scripts                 scripts/scripts.go:107 >        register structure: 'HTTP'
INFO    migrations              migrations/migrations.go:81 >   Applied 0 migrations!
INFO    logging_db              logging_db/logging_db.go:72 >   start
INFO    mqtt                    mqtt/mqtt.go:161 >      Serving MQTT server at tcp://[::]:1883
INFO    scripts                 scripts/scripts.go:107 >        register structure: 'Mqtt'
INFO    mqtt                    server/server.go:864 >  init plugin hook wrappers
INFO    mqtt                    server/server.go:972 >  open persistence succeeded      {"type": "memory"}
INFO    mqtt                    server/server.go:972 >  init session store succeeded    {"type": "memory", "session_total": 0}
INFO    mqtt                    server/server.go:972 >  init queue store succeeded      {"type": "memory", "session_total": 0}
INFO    mqtt                    server/server.go:972 >  init subscription store succeeded       {"type": "memory", "client_total": 0}
INFO    mqtt                    server/server.go:945 >  loading plugin  {"name": "admin"}
INFO    mqtt.admin              admin/admin.go:75 >     loaded ...
INFO    mqtt                    mqtt/mqtt.go:166 >      gmqtt server started    {"tcp server listen on": ["[::]:1883"], "websocket server listen on": []}
INFO    scheduler               scheduler/scheduler.go:109 >    started ...
INFO    mqtt_authenticator      qtt_authenticator/mqtt_authenticator.go:138 >   register ...
INFO    zigbee2mqtt             zigbee2mqtt/bridge.go:76 >      bridge id 1,  base topic: zigbee2mqtt
INFO    mqtt                    mqtt/mqtt.go:259 >      new mqtt client 'bridge_zigbee2mqtt'
INFO    mqtt                    mqtt/client.go:55 >     client bridge_zigbee2mqtt, subscribed to topic 'zigbee2mqtt/bridge/#'
INFO    validation              validation/validation.go:68 >   start ...
INFO    stream                  stream/stream.go:124 >  subscribe event_get_last_state
INFO    stream                  stream/stream.go:124 >  subscribe event_add_webpush_subscription
INFO    stream                  stream/stream.go:124 >  subscribe event_get_webpush_public_key
INFO    stream                  stream/stream.go:124 >  subscribe command_terminal
INFO    logging_ws              logging_ws/logging_ws.go:63 >   start
.......................................................................................................context deadline exceeded

Can you, please, give me a hint what I did wrong. I compiled the project from scratch on a ARM64.

With kindest regards
Joerg

**Comment by e154** [comment]
Hello Joerg,

Thank you for your interest. The error is due to the fact that the server did not have time to boot within a specific time (15 seconds) and the launch was interrupted.

https://github.com/e154/smart-home/blob/6b0b6f7fa65347d97f4b371d7115b4d2afd51c85/common/app/app.go#L36

This value will be increased in the next release

**Comment by joehil** [comment]
Hello Alexey,
thank you for your hint!

I increased the timeout to 60 seconds and recompiled the project. And now the server starts alright, thank you again!

With kindest regards
Joerg
